### ISSUE TITLE
Enhance backend API security by implementing nginx authorization for user access

### ISSUE THREAD

**Issue by raftmsohani** [raft-tech/TANF-app#2998]
## Background ##
Recent ZAP scanner findings revealed that the backend API is accessible via direct browsing, even though unauthorized users are blocked by backend logic. 

While no data is exposed due to authentication checks, the ability to directly browse the backend API presents a potential security risk. To strengthen security, backend API access should be enforced at the **Nginx layer**, preventing unauthorized users from reaching the API routes entirely.

This approach mirrors the TANF project's Nginx configuration:  
https://github.com/raft-tech/TANF-app/blob/0b10e3f32224b5f4a1b70e55528d70e8b32c5693/tdrs-frontend/nginx/cloud.gov/locations.conf#L46

## Why It Matters ##
- Improves overall system security by restricting backend access at the server level
- Prevents unnecessary load on the backend by blocking unauthorized requests earlier
- Addresses OWASP scan findings and strengthens defense-in-depth

## What's Changing ##
- Nginx configuration will be updated to restrict `/v1` API route access to authorized users only
- Unauthorized requests will be rejected with a `403 Forbidden` response directly from Nginx
- Backend logic remains in place but will no longer be the first line of defense

### Acceptance Criteria ###
- [ ] Authorized users can access backend API endpoints at `/v1`
- [ ] Unauthorized users receive a `403 Forbidden` response from Nginx — not the backend API
- [ ] All automated tests pass, including API access checks based on user session and authorization status
- [ ] Testing Checklist is completed and passes
- [ ] README is updated if necessary to document the Nginx configuration change
- [ ] [Release digest](https://app.gitbook.com/o/Yd4Wv0Fi89kSKpmakWrn/s/JSIbkxujKUxhKv2cXFEg/) has been updated with testing steps & relevant media  

### Usability Testing Criteria ###
_N/A — This is a backend security enhancement. Validation occurs through functional testing and security scans._

### Notes ###
- This implementation addresses several lower-priority OWASP findings
- It improves security posture by applying defense-in-depth and moving authorization checks closer to the edge
- Consider follow-up work to review other publicly reachable endpoints that may need similar protections

### Supporting Documentation ###
- Reference TANF Nginx configuration: [locations.conf](https://github.com/raft-tech/TANF-app/blob/0b10e3f32224b5f4a1b70e55528d70e8b32c5693/tdrs-frontend/nginx/cloud.gov/locations.conf#L46)

**Comment by kennymcnett** [comment]
Closing as part of backlog cleanup. This issue has been blocked for 12+ months. If the underlying work is still needed, please create a new issue with current context and blocker status.
